src/runtime/virtcontainers/pkg/cgroups/cgroups_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package cgroups
+
+import (
+	"testing"
+
+	"github.com/containerd/cgroups"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/stretchr/testify/assert"
+)
+
+func newCgroup() *Cgroup {
+	return &Cgroup{
+		cgroup:  &mockCgroup{},
+		path:    "",
+		cpusets: &specs.LinuxCPU{},
+		devices: []specs.LinuxDeviceCgroup{},
+	}
+}
+
+func TestSandboxDevices(t *testing.T) {
+	assert := assert.New(t)
+
+	devices := sandboxDevices()
```

Comment:
Suggestion: check the detail devices like `/dev/kvm`, `/dev/null`, and `wildcardDevices`.

Reply:
Thanks @liubin !